### ISSUE TITLE
Does this work?

### ISSUE THREAD

**Issue by AndreKR** [nioc/node-red-contrib-opentelemetry#1]
I have added this library, added an OTEL node to one of my flows, set the URL and enabled the debug log. Then in a different node I have created a flow consisting of an inject, an http request and a debug node. If I run this flow, I can see this debug output:
```plain
5.onReceive: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: >> dest: inject 6e90cb67d98afd6a
1.onSend: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: src: inject 6e90cb67d98afd6a
=> Created parent span for inject
=> Created span for inject
3.preDeliver: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: src: inject 6e90cb67d98afd6a >> dest: http request 5b4d494fcc8f532a
4.postDeliver: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: src: inject 6e90cb67d98afd6a >> dest: http request 5b4d494fcc8f532a
=> Created span for http request
7.onComplete: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: ## node: inject 6e90cb67d98afd6a
==> Ended span for  6e90cb67d98afd6a inject
6.postReceive: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: >> dest: inject 6e90cb67d98afd6a
5.onReceive: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: >> dest: http request 5b4d494fcc8f532a
6.postReceive: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: >> dest: http request 5b4d494fcc8f532a
1.onSend: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: src: http request 5b4d494fcc8f532a
3.preDeliver: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: src: http request 5b4d494fcc8f532a >> dest: debug 7886ed2c49c9f953
4.postDeliver: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: src: http request 5b4d494fcc8f532a >> dest: debug 7886ed2c49c9f953
7.onComplete: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: ## node: http request 5b4d494fcc8f532a
==> Ended span for  5b4d494fcc8f532a http request
5.onReceive: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: >> dest: debug 7886ed2c49c9f953
7.onComplete: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: ## node: debug 7886ed2c49c9f953
==> Ended span for  7886ed2c49c9f953 debug
Parent span "Message inject" no longer has child span, ending
6.postReceive: rootMsgId: 0624052f09d15428, _msgId: 0624052f09d15428: >> dest: debug 7886ed2c49c9f953
```

But that's it. No HTTP connection is made to the URL (I verified with tcpdump) but also no error message appears. Shouldn't the trace be transmitted at some point?

**Comment by nioc** [comment]
Yes it works on my Node-RED instance (3.0.2) with the 1.0.2 version.

Here is my telemetry flow (only the OTEL node):

```json
[
    {
        "id": "ac125c4c6d10643e",
        "type": "OpenTelemetry",
        "z": "1dc4c3fe2e38934c",
        "name": "",
        "url": "http://jaeger:4318/v1/traces",
        "serviceName": "Node-RED",
        "rootPrefix": "Message",
        "timeout": "10",
        "ignoredTypes": "debug,catch,request id",
        "propagateHeadersTypes": "find actions, create action, create raw action, transmit message, find equipments, update equipment, create equipment, create raw equipment, find states, update state, create state, create raw state",
        "isLogging": false,
        "x": 760,
        "y": 160,
        "wires": []
    }
]
```

I have a Jaeger "All in one" container running on same docker compose (same network):

``` yml
version: "3.4"

services:
  node-red:
    image: nodered/node-red:latest
    restart: unless-stopped
    environment:
      - TZ=Europe/Paris
    ports:
      - "1880:1880"
    volumes:
      - ./node-red/data:/data
  # ...
  jaeger:
    image: jaegertracing/all-in-one:latest
    restart: unless-stopped
    environment:
      - TZ=Europe/Paris
    ports:
      - "16686:16686" #frontend
      - "9411:9411" #http collector in zipkin format
    volumes:
      - jaeger:/tmp
    command:
      - --collector.zipkin.host-port=9411
      - --query.base-path=/jaeger-ui
      - --query.max-clock-skew-adjustment=10s
      - "--collector.otlp.enabled=true"
```

Here a trace:

![Jaeger UI](https://github.com/nioc/node-red-contrib-opentelemetry/assets/5225411/63af5c87-ee60-4d8a-bf5a-fdb5138bb5d9)

Could you provide some information about your Node-RED/OTEL collector setup?

**Comment by AndreKR** [comment]
Node-RED 3.1.0-beta.4 installed with Docker. Projects enabled. As "collector" I'm just using a request bin from https://pipedream.com/requestbin to test if any HTTP request is generated.

**Comment by AndreKR** [comment]
I'm doing some debugging... the exporter does get called and it returns `{ code: 0 }`. I'll investigate further.

**Comment by nioc** [comment]
Just an idea: Try to listen/set an url with standard Jaeger port, as this node use OTEL library, may be it does stuff according to configured port.

**Comment by AndreKR** [comment]
It's working now. I think the issue was that a Node-RED restart was required before the configured URL was used. A deploy was not enough. Naturally during debugging I changed the source and restarted Node-RED, so that's why it's working now.

**Comment by AndreKR** [comment]
I checked again, yes, a restart of Node-RED is required to set the URL.

At least I think the readme should mention this. Even better would of course be if it wasn't necessary at all.

**Comment by nioc** [comment]
16 months later (😆): this need to restart should be fixed by #13
